core/src/main/java/dk/alexandra/fresco/framework/builder/numeric/field/BigIntegerModulus.java

```diff
@@ -0,0 +1,33 @@
+package dk.alexandra.fresco.framework.builder.numeric.field;
+
+import java.io.Serializable;
+import java.math.BigInteger;
+import java.util.Objects;
+
+/**
+ * This class is a naïve implementation of a modulus, other field structures would
+ * have more complicated data structures than this.
+ */
+final class BigIntegerModulus implements Serializable {
+
+  private static final long serialVersionUID = 1L;
+  private final BigInteger value;
+
+  BigIntegerModulus(BigInteger value) {
+    if (value.compareTo(BigInteger.ZERO) <= 0) {
+      throw new IllegalArgumentException("Only positive modulus is acceptable");
```

Comment:
May want to java doc this constructor to say that the value must be larger than zero.